### PR TITLE
Allow psr/log 3

### PR DESCRIPTION
**Pull request by franmomu**
There is https://github.com/elastic/elasticsearch-php/pull/1184 and https://github.com/elastic/elasticsearch-php/pull/1187, but it needs a little bit more of work since there are implementations of psr/log.

### CODE REVIEW

File: src/Elasticsearch/Common/EmptyLogger.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ class EmptyLogger extends AbstractLogger implements LoggerInterface
     /**
      * {@inheritDoc}
      */
-    public function log($level, $message, array $context = [])
+    public function log($level, $message, array $context = []): void
```
Comment (franmomu) [inline src/Elasticsearch/Common/EmptyLogger.php:35]: note that this is BC break, but given the name of the class, I guess it is not meant to be extended.

File: composer.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "php": "^7.3 || ^8.0",
     "ext-json": ">=1.3.7",
     "ezimuel/ringphp": "^1.1.2",
-    "psr/log": "^1|^2"
+    "psr/log": "^1|^2|^3"
```
Comment (cronlabspl) [inline composer.json:22]: PSR-2 is [deprecated](https://www.php-fig.org/psr/psr-2/) in favour of [PSR-12](https://www.php-fig.org/psr/psr-12/)

**Review by cronlabspl — CHANGES_REQUESTED** [review]
PSR-2 is [deprecated](https://www.php-fig.org/psr/psr-2/) in favour of [PSR-12](https://www.php-fig.org/psr/psr-12/)

File: composer.json (hunk 2 above)
Comment (franmomu) [reply composer.json:22]: I think you're mixing things here, [`psr/log`](https://github.com/php-fig/log) package is about [`PSR-3`](https://www.php-fig.org/psr/psr-3/) which currently has 3 versions.

PSR-2 and PSR-12 are about coding standard

Comment (cronlabspl) [reply composer.json:22]: @franmomu correct, but it doesnt change the fact that PSR-2 is deprecated

Comment (franmomu) [reply composer.json:22]: so... what should I change in this PR?

Comment (cronlabspl) [reply composer.json:22]: remove `^2` from line 22 in `composer.json`

Comment (franmomu) [reply composer.json:22]: Please read this again:

> I think you're mixing things here, [`psr/log`](https://github.com/php-fig/log) package is about [`PSR-3`](https://www.php-fig.org/psr/psr-3/) which currently has 3 versions.

`psr/log` has nothing to do with psr-2

Comment (cronlabspl) [reply composer.json:22]: > `psr/log` has nothing to do with psr-2

Yes. So will close this as misunderstanding. Must have been.

**Comment by ruudk** [comment]
@cronlabspl Can you please remove or hide your comment as it's confusing (see thread).

@ezimuel @karyna-tsymbal-atwix @jrodewig Can this PR be merged? Multiple people are now proposing the same change. This change is needed to be able to upgrade to Symfony 6.

**Comment by technige** [comment]
This is a duplicate of #1184 and #1187, the latter of which is now closed as an exact duplicate.

@franmomu We very much appreciate the extra detail you have added to this issue, but please could you instead collaborate on #1184 (the original issue), rather than opening a new one. This will help to expedite our work in reviewing and merging. Once you have do so, this issue can be closed.

**Comment by franmomu** [comment]
@technige no problem merging that one!

I created this one because I was able to run the workflow jobs and the other PRs I thought they were created editing the file from Github directly, so the purpose was in fact to ease the review (avoid approving the workflow, see if the tests passes, make changes, approve again, etc), but as I said no worries at all merging the other one.

**Comment by ruudk** [comment]
This can be closed as it has been merged into https://github.com/elastic/elasticsearch-php/pull/1184

**Pull request closed without merging**